### PR TITLE
Documentation: Update developer guide with AddTagCommand

### LINKED ISSUE #253: Add AddSpecialisation/MedicalCondition/Pescriptions to DG's Implementation Section

### PR DESCRIPTION
**Pull request by mounilsankar**
Closes #253 #259, #268

Also fixed bug where invalid dates with correct format was allowed

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -479,140 +482,42 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+The example usage scenario for the add prescription and add medical condition mechanisms would be very similar to the above scenario.
 
-<img src="images/SortActivityDiagram.png" />
-
-#### Design considerations:
+The following sequence diagram shows how the add specialisation operation would work:
+![AddSpecialisationSequenceDiagram](images/AddSpecialisationSequenceDiagram.png)
 
-**Aspect: How to sort the filteredLists in Database:**
+The sequence diagram for the add specialisation and medical condition operations would be similar
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:516]: `specialisation` change to `prescription`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -254,13 +250,15 @@ The following activity diagram summarizes what happens when a user wants to find
 
 **Introduction**
 
-The listing of all doctors/patient in the database is facilitated by 'LogicManager'. It extends 'Logic' and stores the mediConnectParser that parses the user input, and the model in which the command is executed. Additionally, it implements the following operations:
+This section describes the feature that allows users to list doctors/patient in the MediConnect database.
```
[the remark this answers is not in the document]
Comment (mounilsankar) [reply docs/DeveloperGuide.md:253]: this one should be list i believe

[the remark this answers is not in the document]
Comment (yanghengtang) [reply docs/DeveloperGuide.md:253]: oh yes i see wrong

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -479,140 +482,42 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Step 2. The user inputs `list_d `. MediConnect will display the FilteredDoctorList in its default sorting order.
+Step 3. The user inputs `add_tag_d 2 t\Orthopaedic`  to add the prescription 'Orthopaedic' to the doctor at index 2 in the displayed doctor list.
+The `add_tag_d` command calls `AddSpecialisationCommandParser#parse` which parses the index argument which is the index of the doctor to delete as well as the tag argument which is the tag to be added to the doctor at the specified index.
+A new `AddSpecialisationCommand` instance will be created
 
-The sort doctor and sort patient functionality works in a similar manner. It selects the relevant `Comparator`, constructs the `ListDoctorCommand`/`ListPatientCommand`,
-and sorts the corresponding `FilteredList` within `Database`.
+Step 4. The created `AddSpecialisationCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`AddSpecialisationCommand#execute` then calls `Model#setDoctor` and with the given `Index` and the doctor with the updated prescription.
+The tag 'Orthopaedic' is then added to the' `Doctor` at the `Index` in the filteredDoctorList by calling `FilteredList#setDoctor`.
```
[the remark this answers is not in the document]
Comment (mounilsankar) [reply docs/DeveloperGuide.md:508]: but its the setDoctor in the FilteredList that modifies the doctor tho. But its missing  a few steps in between

**Review by yanghengtang — APPROVED** [review]

**Pull request merged**